MyApplication/app/src/main/java/net/serkanbal/myapplication/MainActivity.java

```diff
@@ -0,0 +1,139 @@
+package net.serkanbal.myapplication;
+
+import android.app.job.JobInfo;
+import android.app.job.JobScheduler;
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.support.v4.content.LocalBroadcastManager;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.TextView;
+
+public class MainActivity extends AppCompatActivity {
+    TextView mJob1Before, mJob1After, mJob2Before, mJob2After, mJob3Before, mJob3After;
+    BroadcastReceiver mReceiver, mReceiver2, mReceiver3;
+    Integer mFirstNumber = 0, mSecondNumber = 0;
+    String mFirstDate = "", mSecondDate = "", mFirstToken = "", mSecondToken = "";
```

Comment:
With no access modifier specified, the access defaults to "package private" meaning it's accessible to any classes within the same package. That means that if I was on your team, and I added a RecyclerView to the activity, in the RecyclerView Adapter in `onBindView()` i could so something like `((MainActivity) holder.getContext()).mJob1Before.setText("Hi Serkan!");` You probably don't want to let me do that!

The [official docs on access modifiers](http://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html) say "Use the most restrictive access level that makes sense for a particular member. Use private unless you have a good reason not to." Scroll to the bottom for their tips.

Here's another thread that discuss this idea: http://softwareengineering.stackexchange.com/questions/314192/best-practices-for-using-public-protected-private